config/daemonconfig/daemonconfig.go

```diff
@@ -136,6 +159,142 @@ func DumpConfigString(c interface{}) (string, error) {
 	return string(b), err
 }
 
+func IsDirExisted(path string) bool {
+	s, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return s.IsDir()
+}
+
+func makeStringSlice(slice []interface{}, cb func(string) string) ([]string, error) {
+	out := make([]string, len(slice))
+	for i, value := range slice {
+		str, ok := value.(string)
+		if !ok {
+			return nil, fmt.Errorf("unable to cast %v to string", value)
+		}
+
+		if cb != nil {
+			out[i] = cb(str)
+		} else {
+			out[i] = str
+		}
+	}
+	return out, nil
+}
+
+func parseMirrorsConfigFromToml(b []byte) ([]MirrorConfig, error) {
```

Comment:
Please also move this function to the mirrors.go

Reply:
Done.